PYME/IO/csv_flavours.py

```diff
@@ -0,0 +1,229 @@
+                    delim = guess_delim(line,delims)
+                dataLines.append(line.split(delim))
+                n += 1
+                
+        numCommentLines = len(commentLines)
+        numCols = len(dataLines[0])
+        numHeaderNameLines = len(headerNameLines)
+        
+        if len(headerNameLines) > 0 and len(headerNameLines[-1].split(delim)) == numCols:
+            colNamesRaw = [s.strip() for s in headerNameLines[-1].split(delim)]
+            # the stuff below seemed necessary since (1) some names came with byte order mark, or BOM, prepended
+            # and (2) some had quotes around the names
+            colNames = [name.encode('utf-8').decode('utf-8-sig').strip('"') for name in colNamesRaw]
+        else:
+            colNames = ['column_%d' % i for i in range(numCols)]
+
+        self.colNames = colNames
+        self.dataLines = dataLines
+        self.nHeaderLines = numCommentLines + numHeaderNameLines
+        self.guessedDelim = delim
+        self.commentLines = commentLines
+
+
+    def replace_names(self):
+        newnames = []
+        if self.flavour == 'default':
+            repdict = { # a few default translations, just in case
+                'X' : 'x',
+                'Y' : 'y',
+                'Z' : 'z',
+            }
+        else:
+            repdict = self.csv_flavours[self.flavour]['column_name_mappings']
+
+        for name in self.colNames:
+            if name in repdict.keys():
+                newname = repdict[name]
+            else:
+                newname = name
+            newnames.append(newname.replace(' ','_')) # in names we replace spaces with underscores
+        self.translatedNames = newnames
+
+
+    def check_required_names(self):
+        reqNotDef = [name for name in self.requiredNames.keys() if not name in self.translatedNames]
+        if len(reqNotDef) > 0:
+            raise RuntimeError("some required names are not defined in file header: " + repr(reqNotDef))
+            # this is a stopgap; the proper implementation will need to call into the textimportdialog
+            # at this stage
+
+            
+    def read_csv_data(self):
+        return np.genfromtxt(self.filename,
+                             comments=self.flavour_value_or_default('comment_char','#'),
+                             delimiter=self.flavour_value_or_default('delimiter',','),
+                             skip_header=self.nHeaderLines,
+                             skip_footer=self.flavour_value_or_default('skip_footer',0),
+                             names=self.translatedNames, dtype='f4', replace_space='_',
+                             missing_values=None, filling_values=np.nan, # use NaN to flag missing values
+                             invalid_raise=not self.flavour_value_or_default('must_ignore_errors', False),
+                             encoding='latin-1') # Zeiss Elyra bombs unless we go for latin-1 encoding, maybe make flavour specific?
```

Comment:
@csoeller - could you please provide an example Elyra file? It would be nice to double check if any alternate encodings work (UTF8 would be ideal).

Reply:
For that purpose I have created a githuib repo with test files and also a small test suite, see https://github.com/csoeller/PYME-SMLM-import-tests.

This repo includes example files for all the flavours I have tried to cover so far, including [elyra examples](https://github.com/csoeller/PYME-SMLM-import-tests/tree/main/elyra). The test rely on the ability of the programmatic opening of these files without gui dialog via a `pipeline` constructor command, e.g.

```python
    pl = pipeline.Pipeline(filename=fname)
```

It would be good to preserve this ability, possibly with a `noGUI` or similar keyword if required.

Reply:
yeah, I realised shortly after I'd written that that you'd already mentioned the example data earlier - sorry about that.

Reply:
Have just looked at the example elyra file -  is the kind of corruption (see below) that requires the `invalid_raise=False` option common, or is it limited to this one file?

```
40146	1486	1	0	2195.5	35172.0	23.3	885	47.42	0.65	173.4	1	1
40147	1486	1	0	35413.4	40391.3	30.0	446	71.18	0.68	131.2	1	1
40148	1486	1	0	29306.8	50081.7	29.6	395	43.82	0.59	134.3	1	1
40149	1486	1	0	31793.5	42414.7	28.6	569	58.19	0.74	148.0	1	1
40150	1486	1	0	31947.4	55524.0	25.3	549	40.44	0.80	141.8	1	1
40151	1486	1	0	31913.2	40789.0	25.6	815	45.05	0.63	176.4	1	1
40152	1486

												
40155	1486	1	0	35894.0	35253.5	37.2	289					
40156	1486	1	0	14841.3	41790.8	35.8	254	52.38	0.68	114.6	1	1
40157	1486	1	0	32679.2	34909.4	37.0	583	75.16	0.68	166.0	1	1
40158	1486	1	0	30516.5	54053.7	34.2	449	68.38	0.69	141.7	1	1
40159	1486	1	0	37139.9	31912.4	28.1	75	54.00	0.69	45.2	1	1
40160	1487	1	0	16720.8	35844.3	9.2	2268	64.80	0.89	150.2	1	1
```

If it's a one-off problem with a specific corrupted file, then we should possibly not just ignore it. If it's common occurrence for elyra files, I'd be happier ignoring.